clippy_lints/src/disallowed_script_idents.rs

```diff
@@ -89,9 +89,13 @@ impl EarlyLintPass for DisallowedScriptIdents {
             // Fast path for ascii-only idents.
             if !symbol_str.is_ascii()
                 && let Some(script) = symbol_str.chars().find_map(|c| {
-                    c.script_extension()
-                        .iter()
-                        .find(|script| !self.whitelist.contains(script))
+                    (!c.is_ascii())
+                        .then(|| {
+                            c.script_extension()
+                                .iter()
+                                .find(|script| !self.whitelist.contains(script))
+                        })
+                        .flatten()
```

Comment:
Seems like using an if block is much cleaner here than a nested `.then(.find)` with a flatten.

Or use `filter` before `find_map`